tracer/src/Datadog.Trace/Ci/TestModule.cs

```diff
@@ -41,45 +47,115 @@ private TestModule(string name, string? framework, string? frameworkVersion, Dat
+                OSVersion = CIVisibility.GetOperatingSystemVersion(),
+                CiEnvVars = sessionSpanTags.CiEnvVars,
+                SessionId = sessionSpanTags.SessionId,
+                Command = sessionSpanTags.Command,
+                WorkingDirectory = sessionSpanTags.WorkingDirectory,
+            };
+        }
+        else
         {
-            tags.CiEnvVars = Vendors.Newtonsoft.Json.JsonConvert.SerializeObject(variablesToBypass);
+            // Out-of-Proc session
+            tags = new TestModuleSpanTags
+            {
+                Type = TestTags.TypeTest,
+                Module = name,
+                Framework = framework,
+                FrameworkVersion = frameworkVersion,
+                CIProvider = environment.Provider,
+                CIPipelineId = environment.PipelineId,
+                CIPipelineName = environment.PipelineName,
+                CIPipelineNumber = environment.PipelineNumber,
+                CIPipelineUrl = environment.PipelineUrl,
+                CIJobName = environment.JobName,
+                CIJobUrl = environment.JobUrl,
+                StageName = environment.StageName,
+                CIWorkspacePath = environment.WorkspacePath,
+                GitRepository = environment.Repository,
+                GitCommit = environment.Commit,
+                GitBranch = environment.Branch,
+                GitTag = environment.Tag,
+                GitCommitAuthorDate = environment.AuthorDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
+                GitCommitAuthorName = environment.AuthorName,
+                GitCommitAuthorEmail = environment.AuthorEmail,
+                GitCommitCommitterDate = environment.CommitterDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
+                GitCommitCommitterName = environment.CommitterName,
+                GitCommitCommitterEmail = environment.CommitterEmail,
+                GitCommitMessage = environment.Message,
+                BuildSourceRoot = environment.SourceRoot,
+                LibraryVersion = TracerConstants.AssemblyVersion,
+                RuntimeName = frameworkDescription.Name,
+                RuntimeVersion = frameworkDescription.ProductVersion,
+                RuntimeArchitecture = frameworkDescription.ProcessArchitecture,
+                OSArchitecture = frameworkDescription.OSArchitecture,
+                OSPlatform = frameworkDescription.OSPlatform,
+                OSVersion = CIVisibility.GetOperatingSystemVersion(),
+            };
+
+            if (environment.VariablesToBypass is { } variablesToBypass)
+            {
+                tags.CiEnvVars = Vendors.Newtonsoft.Json.JsonConvert.SerializeObject(variablesToBypass);
+            }
+
+            // Extract session variables (from out of process sessions)
+            var environmentVariables = Environment.GetEnvironmentVariables();
```

Comment:
do you need a wrapper around this to catch exceptions?

Reply:
True, done in https://github.com/DataDog/dd-trace-dotnet/pull/3389/commits/56ad336fe298826a71d551921e735f3be06c1509